src/Features/LanguageServer/Protocol/Handler/Diagnostics/PullDiagnosticCategories.cs

```diff
@@ -22,7 +22,9 @@ internal static class PullDiagnosticCategories
 
         // Fine-grained document pull categories to allow diagnostics to more quickly reach the user.
 
-        public const string DocumentCompilerSyntax = nameof(DocumentCompilerSyntax);
+        // alternatively, VSInternalDiagnosticKind.Syntax.Value
```

Comment:
consider doc'ing why it's essential that we actually use this exact text.  namely that the VS side looks for this.